source/_integrations/ecovacs.markdown

```diff
@@ -12,20 +12,28 @@ ha_codeowners:
   - '@edenhaus'
 ha_domain: ecovacs
 ha_platforms:
+  - binary_sensor
   - vacuum
 ha_integration_type: integration
 ---
 
-The `ecovacs` integration is the main integration to integrate all [Ecovacs](https://www.ecovacs.com) (Deebot) vacuums. You will need your Ecovacs account information (username, password) to discover and control vacuums in your account.
-
-There is currently support for the following device types within Home Assistant:
-
-- [Vacuum](#vacuum)
+The `ecovacs` {% term integration %} is the main integration to integrate all [Ecovacs](https://www.ecovacs.com) (Deebot) vacuums. You will need your Ecovacs account information (username, password) to discover and control vacuums in your account.
 
 {% include integrations/config_flow.md %}
 
 Additional note: There are some issues during the password encoding. Using some special characters (e.g., `-`) in your password does not work.
 
+## Provided entities
+
+The Ecovacs integration provides a vacuum {% term entity %} for each device that is connected to your Ecovacs account.
+
+Using the vacuum entity, you can monitor and control your Ecovacs Deebot vacuum.
+
+Additionally, **depending on your model** the integration provides the following entities:
+
+- **Binary sensor**: 
+  - `Mop attached`: On if the mop is attached. Some models reports a state change only if the state of the vacuum has changed. 
```

Comment:
I don't understand. You are saying a state change is reported only if the state has changed. Isn't that expected behavior? I'm probably missing something here.

Reply:
Some models report a state change of the mop attached sensor only when the vacuum entity's state has changed.

For example:
- Vacuum is docked with the mop attached
- You remove the mop (no update will be pushed to HA)
- You start a new cleaning so the state of the vacuum changes to `cleaning`, and also the binary sensor state update to no mop attached.
I assume that the robot has some sort of sleep mode, where he will not check if a mop is attached or not